website/docs/developer/research/research-APIGateway.md

```diff
@@ -0,0 +1,73 @@
+---
+id: research-APIGateway
+title: API Gateway Research Notes
+description: Research Notes on API Gateway.
+---
+
+## Overview
+
+[AWS API Gateways](https://aws.amazon.com/api-gateway/) are used to define access to a service.
+
+### Implementation Details
+- Requires several components in Terraform.
+  - aws_api_gateway_rest_api
+    ```
+    resource "aws_api_gateway_rest_api" "[name]_api" {
+      name = "request_status_[name]_api"
+    }
+    ```
+  - aws_api_gateway_resource
+    ```
+    resource "aws_api_gateway_resource" "request_status_[name]_api_resource" {
+      path_part   = "resource"
+      parent_id   = aws_api_gateway_rest_api.request_status_[name]_api.root_resource_id
+      rest_api_id = aws_api_gateway_rest_api.request_status_[name]_api.id
+    }
+    ```
+  - aws_api_gateway_method
+    ```
+    resource "aws_api_gateway_method" "request_status_[name]_api_method" {
+      rest_api_id   = aws_api_gateway_rest_api.request_status_[name]_api.id
+      resource_id   = aws_api_gateway_resource.request_status_[name]_api_resource.id
+      http_method   = "GET"
+      # todo: Make sure this is locked down against external access.
+      authorization = "NONE"
+    }
+    ```
+  - aws_api_gateway_integration
+    ```
+    resource "aws_api_gateway_integration" "integration" {
+      rest_api_id             = aws_api_gateway_rest_api.request_status_[name]_api.id
+      resource_id             = aws_api_gateway_resource.request_status_[name]_api_resource.id
+      http_method             = aws_api_gateway_method.request_status_[name]_api_method.http_method
+      integration_http_method = "POST"
+      type                    = "AWS_PROXY"
+      uri                     = aws_lambda_function.request_status_[name].invoke_arn
+    }
+    ```
+  - aws_lambda_permission
+    ```
+    resource "aws_lambda_permission" "request_status_[name]_api_permission" {
+      statement_id  = "AllowExecutionFromAPIGateway"
+      action        = "lambda:InvokeFunction"
+      function_name = aws_lambda_function.lambda.function_name  # todo: Pass this in
+      # todo: Could make this the accountID. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_permission
+      principal     = "apigateway.amazonaws.com"
+    
+      # More: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+      source_arn = "arn:aws:execute-api:${var.region}:${var.accountId}:${aws_api_gateway_rest_api.request_status_[name]_api.id}/*/$    {aws_api_gateway_method.request_status_[name]_api_method.http_method}${aws_api_gateway_resource.request_status_[name]_api_resource.path}"
```

Comment:
According to this [example](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_permission#specify-lambda-permissions-for-api-gateway-rest-api), this could also be simplified to: 

"${aws_api_gateway_rest_api.[name]_api.execution_arn}/*/$    {aws_api_gateway_method.request_status_[name]_api_method.http_method}${aws_api_gateway_resource.request_status_[name]_api_resource.path}"